gio/src/task.rs

```diff
@@ -0,0 +1,121 @@
+use crate::AsyncResult;
+use crate::Cancellable;
+use crate::Task;
+use glib::object::IsA;
+use glib::translate::*;
+use std::boxed::Box as Box_;
+
+impl Task {
+    pub fn new<P: IsA<Cancellable>, Q: FnOnce(&AsyncResult, Option<&glib::Object>) + 'static>(
```

Comment:
This looks like it should also get a futures variant that can be awaited on

Reply:
Well, the idea is more to instantiate a `Task` object and pass it to async functions/futures. When the future will complete, it will call `Task::return_int` or similar and this will trigger the callback associated to the `Task`. Inside the callback typically there will be a call to `Task::propagate_int()` to get the result of the task.
All this is just to be able to implement "old-style" APIs using the async/finish machinery (which requires GTasks and GAsyncResults) fully in Rust.

Is this what you meant / does this address your point? :)

Reply:
Can you provide an example of how all that would work in the `examples` subdirectory? :)

Reply:
sure, I'm adding the example, I will soon update the PR with it